src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+        Map<String, Integer> reports = new HashMap<String, Integer>();
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String value = reader.readLine();
+
+            while (value != null) {
+                String[] values = value.split(",");
+                String product = values[0];
+
+                int count = Integer.parseInt(values[1]);
+
+                reports.merge(product, count, Integer::sum);
+
```

The catch block for reading does `throw new RuntimeException("Can't read the file");` which discards the original exception. This violates the checklist item: "Don't ignore exceptions Leaving an empty catch block or `e.printStackTrace` here is a bad practice. Better re-throw `RuntimeException` with the original exception and some message in the parameters:". Include the original exception as the cause and a contextual message, e.g. `throw new RuntimeException("Can't read data from file " + fromFileName, e);`.